gpkit/constraints/set.py

```diff
@@ -127,19 +128,17 @@ def subinplace(self, subs, value=None):
         for constraint in self:
             constraint.subinplace(subs, value)
 
-    @property
-    def varkeys(self):
-        "return all Varkeys present in this ConstraintSet"
-        return self._varkeys()
-
-    def _varkeys(self, init_dict=None):
-        "return all Varkeys present in this ConstraintSet"
-        init_dict = {} if init_dict is None else init_dict
-        out = KeySet(init_dict)
+    def reset_varkeys(self, init_dict=None):
+        "Goes through constraints and collects their varkeys."
+        varkeys = KeySet()
+        if not init_dict is None:
+            varkeys.update(init_dict)
         for constraint in self:
             if hasattr(constraint, "varkeys"):
-                out.update(constraint.varkeys)
-        return out
+                varkeys.update(constraint.varkeys)
+            # else:
+            #     print type(constraint) # getting numpy bools...
+        self.varkeys = varkeys
```

Comment:
Try the new d8, just pushed.

On Thu, Aug 4, 2016 at 2:07 PM, Warren Hoburg <notifications@github.com>
wrote:

> In gpkit/constraints/set.py
> <https://github.com/hoburg/gpkit/pull/787#discussion_r73602863>:
>
> >          for constraint in self:
> >              if hasattr(constraint, "varkeys"):
> > -                out.update(constraint.varkeys)
> > -        return out
> > +                varkeys.update(constraint.varkeys)
> > +            # else:
> > +            #     print type(constraint) # getting numpy bools...
> > +        self.varkeys = varkeys
>
> and that appears to be true even as of the latest merge commit
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/hoburg/gpkit/pull/787/files/0f425f6e3ba7a2bf56b596de7ca9452334bb1947#r73602863>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABagGNeJA8l4i8ReNpE1dsufFf3bVPgxks5qclR3gaJpZM4Jcvdl>
> .
>